rust/feature-flags/src/handler/flags.rs

```diff
@@ -10,29 +10,181 @@ use crate::{
+#[serde(rename_all = "lowercase")]
+pub enum EvaluationRuntime {
+    All,
+    Client,
+    Server,
+}
+
+impl From<String> for EvaluationRuntime {
+    fn from(s: String) -> Self {
+        EvaluationRuntime::from(s.as_str())
+    }
+}
+
+impl From<&str> for EvaluationRuntime {
+    fn from(s: &str) -> Self {
+        if s.eq_ignore_ascii_case("client") {
+            EvaluationRuntime::Client
+        } else if s.eq_ignore_ascii_case("server") {
+            EvaluationRuntime::Server
+        } else {
+            EvaluationRuntime::All
+        }
+    }
+}
+
+/// Determines the evaluation runtime based on request characteristics.
+/// Uses explicit runtime if provided, otherwise analyzes user-agent, headers, and lib_version
+/// to detect if the request is from a client-side (browser/mobile) or server-side SDK.
+fn detect_evaluation_runtime_from_request(
+    headers: &axum::http::HeaderMap,
+    lib_version: Option<&str>,
+    explicit_runtime: Option<EvaluationRuntime>,
+) -> Option<EvaluationRuntime> {
+    // Use explicitly passed runtime if available
+    if let Some(runtime) = explicit_runtime {
+        return Some(runtime);
+    }
+
+    // Analyze User-Agent header
+    if let Some(user_agent) = headers.get("user-agent").and_then(|v| v.to_str().ok()) {
+        // Browser patterns - these typically indicate client-side execution
+        if user_agent.contains("Mozilla/")
+            || user_agent.contains("Chrome/")
+            || user_agent.contains("Safari/")
+            || user_agent.contains("Firefox/")
+            || user_agent.contains("Edge/")
+        {
+            return Some(EvaluationRuntime::Client);
+        }
+
+        // Server SDK patterns - these indicate server-side execution
+        if user_agent.starts_with("posthog-python/")
+            || user_agent.starts_with("posthog-ruby/")
+            || user_agent.starts_with("posthog-php/")
+            || user_agent.starts_with("posthog-java/")
+            || user_agent.starts_with("posthog-go/")
+            || user_agent.starts_with("posthog-node/") // Note: server-side Node.js
+            || user_agent.contains("python-requests/")
+            || user_agent.contains("curl/")
+        {
+            return Some(EvaluationRuntime::Server);
+        }
+    }
+
+    // Analyze lib_version for additional clues
+    if let Some(lib_version) = lib_version {
+        // JavaScript SDK versions often indicate client-side
+        if lib_version.contains("posthog-js") || lib_version.contains("javascript") {
+            return Some(EvaluationRuntime::Client);
+        }
+
+        // Server SDK patterns in lib_version
+        if lib_version.contains("python")
+            || lib_version.contains("ruby")
+            || lib_version.contains("php")
+            || lib_version.contains("java")
+            || lib_version.contains("golang")
+            || lib_version.contains("server")
+        {
+            return Some(EvaluationRuntime::Server);
+        }
```

Comment:
This seems off to me. I know the SDKs send `lib_version` in event properties (in which case it's just the version number because `lib` contains the library name). But I don't know of any SDK sending `lib_version` in the query string? Is this something we want to start sending?

Reply:
ah, you're so right.  Great catch. 

I got ahead of myself here.  I'm gonna cut this bit altogether; I think we can get enough info from the user agent here.